src/main/dialogs/captureToolDialog.js

```diff
@@ -0,0 +1,476 @@
+            }
+        );
+    }
+};
+
+/*
+ * Generates and returns a set of diagnostics for the current local machine.
+ *
+ * Returns a promise containing these.
+ */
+gpii.app.captureTool.generateDiagnostics = function (diagnosticsCollector) {
+    return fluid.promise.fireTransformEvent(diagnosticsCollector.events.onCollectDiagnostics, {});
+};
+
+/**
+ * Fetches the list of locally installed solutions and send them to the renderer
+ * on the `channelNotifier`.
+ *
+ * @param {gpii.app.channelNotifier} channelNotifier - The channel notifier to the renderer process.
+ * @param {gpii.flowManager.local} flowManager - The local flow manager.
+ */
+gpii.app.captureTool.channelGetInstalledSolutions = function (channelNotifier, flowManager) {
+    flowManager.capture.getInstalledSolutionsForCurrentDevice().then(
+        function (data) {
+            channelNotifier.events.sendingInstalledSolutions.fire(data);
+            var runningSolutions = {};
+            fluid.each(data, function (solutionEntry, solutionId) {
+                try {
+                    fluid.log("ABOUT TO CHECK PROCESS FOR: ", solutionId);
+                    var isRunningState = gpii.processReporter.handleIsRunning(solutionEntry);
+                    fluid.log("RUNNING STATE: ", isRunningState);
+                    if (isRunningState) {
+                        runningSolutions[solutionId] = solutionEntry;
+                    }
+                } catch (err) {
+                    fluid.log("Exception trying to look up", solutionEntry, " Error: ", err);
+                }
+            });
+            channelNotifier.events.sendingRunningSolutions.fire(runningSolutions);
+        },
+        function (err) {
+            channelNotifier.events.sendingInstalledSolutions.fire({isError: true, message: err});
+        }
+    );
+};
+
+/**
+ * This performs the task of capturing solutions using the flow manager, one solution at a time.
+ * While the flow manager can handle capturing multiple solutions in one go, because there is a
+ * likelihood of unknown situations in the real world with solutions and settings handlers, we go
+ * one at a time such that if there is a failure or exception thrown in one solution, it only
+ * affects that solutions. It causes no real discernable notice in speed.
+ *
+ * @param {gpii.flowManager.local} flowManager - The local flow manager.
+ * @param {Array} solutionsList - Array of standard solution ID's.
+ * @return {fluid.promise} A promise resolving to the full settings capture of all the solutions
+ * in `solutionsList`.
+ */
+gpii.app.captureTool.safelyGetSolutionsCapture = function (flowManager, solutionsList) {
+    var promTogo = fluid.promise();
+    // Due to the unknown stability of some settings handlers, we are invoking the system capture
+    // on each solution individually, so if one fails the capture will still proceed.
+    var capturePromises = [];
+    fluid.each(solutionsList, function (solutionId) {
```

Comment:
This is actually fluid.transform, not an each with a side-effect ....

Reply:
Moved this to use fluid.transform.